src/chaos-experiments/models/chaosExperimentsManager.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const httpContext = require('express-http-context'),
+    uuid = require('uuid');
+
+const logger = require('../../common/logger'),
+    databaseConnector = require('./database/databaseConnector'),
+    { ERROR_MESSAGES, CONTEXT_ID } = require('../../common/consts'),
+    generateError = require('../../common/generateError');
+
+module.exports.createChaosExperiment = async function (chaosExperiment) {
+    const contextId = httpContext.get(CONTEXT_ID);
+
+    const chaosExperimentWithTheSameName = await databaseConnector.getChaosExperimentByName(chaosExperiment.name, contextId);
+    if (chaosExperimentWithTheSameName) {
+        throw generateError(400, ERROR_MESSAGES.CHAOS_EXPERIMENT_NAME_ALREADY_EXIST);
+    }
+    const experimentId = uuid.v4();
+    try {
+        await databaseConnector.insertChaosExperiment(experimentId, chaosExperiment, contextId);
+        chaosExperiment.id = experimentId;
```

Comment:
why do you need it here? I would set the id before sending to db connector, this way you have to set it twice

Reply:
I think it's more consistent to put it here - just looked on other resources. don't have a strong opinion here